xarray/core/ewm.py

```diff
@@ -0,0 +1,31 @@
+from numbagg.moving import ewm_nanmean
+
+from .computation import apply_ufunc
+
+
+def _ewm_nanmean(array, com):
+    # wrapper b/c of kwarg
+    # potentially this fuction should be in numbagg?
+    return ewm_nanmean(array, com)
+
+
+class EWM(object):
+    _attributes = ['com', 'dim']
+
+    def __init__(self, obj, spans):
+        # TODO: add alternatives to com
+        self.obj = obj
+        dim, span = next(iter(spans.items()))
+        self.dim = dim
+        self.span = span
+
+
+class DataArrayEWM(EWM):
+    def mean(self):
+        return apply_ufunc(
+            _ewm_nanmean,
+            self.obj,
+            input_core_dims=[[self.dim]],
+            output_core_dims=[[self.dim]],
```

Comment:
Is this the correct approach for a function that:
- Needs to put a dimension in last place, given the gufunc interface
- Needs to retain that dimension (i.e. unlike `.sum` it shouldn't be reduced)

I looked for comparisons, but it looks like rolling functions are pre-`xarray_ufunc` era

Currently, the array
- loses the time coords
- doesn't transpose back

Reply:
Or is this not a fit for `xarray_ufunc` and I should do it manually (which would be easy, but trying to use our tools...)

Reply:
In other cases we have achieved this by adding `.transpose(*self.dims)` after using `apply_ufunc()`. This would work at least for DataArray objects.

The other option would be to drop `input_core_dims`/`output_core_dims` entirely and pass an `axis` argument, e.g.,
```python
        axis = self.get_axis_num(self.dim)
        return apply_ufunc(
            _ewm_nanmean,
            self.obj,
            kwargs=dict(com=self.span, axis=axis)
        )
```

This works out fine because with a single argument there isn't a meaningful difference between "broadcast" and "core" dimensions -- there is no broadcasting that happens.

Reply:
I guess the notion of a "core" dimension does matter if we want to support parallelization in dask arrays with `dask='parallelized'. But at long as you aren't explicitly setting that in `apply_ufunc()` it doesn't make a difference.